core/src/main/java/com/graphhopper/routing/DefaultWeightingFactory.java

```diff
@@ -75,7 +75,7 @@ public Weighting createWeighting(Profile profile, PMap requestHints, boolean dis
                 throw new IllegalArgumentException("custom weighting requires a CustomProfile but was profile=" + profile.getName());
             CustomModel queryCustomModel = requestHints.getObject(CustomModel.KEY, null);
             CustomProfile customProfile = (CustomProfile) profile;
-            if (queryCustomModel != null)
+            if (queryCustomModel != null && !requestHints.getBool("lm.disable", false))
```

Comment:
The problem is that if LM is activated on the server the client would still need the lm.disable=true parameter. Will see if I can detect this with LMProfile.